paasta_tools/cli/cmds/secret.py

```diff
@@ -0,0 +1,172 @@
+    )
+    secret_parser.set_defaults(command=paasta_secret)
+
+
+def print_paasta_helper(secret_path, secret_name):
+    print("\nYou have successfully encrypted your new secret and it\n"
+          "has been stored at {}\n"
+          "To use the secret in a service you can add it to your PaaSTA service\n"
+          "as an environment variable.\n"
+          "You do so by referencing it in the env dict in your yaml config:\n\n"
+          "main:\n"
+          "  cpus: 1\n"
+          "  env:\n"
+          "    MY_SECRET_VAR: SECRET({})\n\n"
+          "Once you have referenced the secret you must commit the newly\n"
+          "created/updated json file and your changes to your yaml config. When\n"
+          "you push to master PaaSTA will bounce your service and the new\n"
+          "secrets plaintext will be in the environment variable you have\n"
+          "specified.".format(secret_path, secret_name))
+
+
+def get_plaintext_input(args):
+    if args.stdin:
+        plaintext = sys.stdin.buffer.read()
+    elif args.plain_text:
+        plaintext = args.plain_text.encode('utf-8')
+    else:
+        print("Please enter the plaintext for the secret, press Ctrl-D when done.")
+        lines = []
+        while True:
+            try:
+                line = input()
+            except EOFError:
+                break
+            lines.append(line)
+        plaintext = '\n'.join(lines).encode('utf-8')
+    return plaintext
+
+
+def is_service_folder(soa_dir, service_name):
+    return os.path.isfile(os.path.join(soa_dir, service_name, "service.yaml"))
+
+
+def _get_secret_provider_for_service(service_name, cluster_names=None):
+    if not is_service_folder(os.getcwd(), service_name):
+        paasta_print(
+            "You must run this tool from the root of your local yelpsoa checkout\n"
+            "The tool modifies files in yelpsoa-configs that you must then commit\n"
+            "and push back to git.",
+        )
+        sys.exit(1)
+    system_paasta_config = load_system_paasta_config()
+    secret_provider_kwargs = {
+        'vault_cluster_config': system_paasta_config.get_vault_cluster_config(),
+    }
+    clusters = cluster_names.split(',') if cluster_names else list_clusters(service_name, os.getcwd())
+    return get_secret_provider(
+        secret_provider_name=system_paasta_config.get_secret_provider_name(),
+        soa_dir=os.getcwd(),
+        service_name=service_name,
+        cluster_names=clusters,
+        secret_provider_kwargs=secret_provider_kwargs,
+    )
+
+
+def paasta_secret(args):
+    secret_provider = _get_secret_provider_for_service(args.service, cluster_names=args.clusters)
+    if args.action in ["add", "update"]:
+        secret_provider.write_secret(
+            action=args.action,
+            secret_name=args.secret_name,
+            plaintext=get_plaintext_input(args),
+        )
+        secret_path = os.path.join(os.getcwd(), args.service, "secrets", "{}.json".format(args.secret_name))
```

Comment:
Is there any place where this path construction is already defined and can be reused?

Reply:
Not entirely but it can be simplified a bit 👍